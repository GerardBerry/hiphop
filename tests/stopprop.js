"use hopscript"

var hh = require("hiphop")

var m =
    <hh.reactivemachine name="foo">
      <hh.outputsignal name="O"/>
      <hh.loop>
	<hh.emit signal_name="O"/>
	<hh.pause/>
      </hh.loop>
    </hh.reactivemachine>;

m.addEventListener("O", function(evt) {
   console.log("first", evt);
   evt.signal = "foo bar";
});

m.addEventListener("O", function(evt) {
   evt.stopPropagation();
   console.log("second", evt);
});

m.addEventListener("O", function(evt) {
   console.log("third", evt);
});

exports.prg = m;
