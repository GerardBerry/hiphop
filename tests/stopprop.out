foo> ;
first { machine: 'foo',
  signal: 'O',
  stopPropagation: [Function],
  state: { stop: false } }
second { machine: 'foo',
  signal: 'O',
  stopPropagation: [Function],
  state: { stop: true } }
--- Output: O
foo> ;
first { machine: 'foo',
  signal: 'O',
  stopPropagation: [Function],
  state: { stop: false } }
second { machine: 'foo',
  signal: 'O',
  stopPropagation: [Function],
  state: { stop: true } }
--- Output: O
foo> ;
first { machine: 'foo',
  signal: 'O',
  stopPropagation: [Function],
  state: { stop: false } }
second { machine: 'foo',
  signal: 'O',
  stopPropagation: [Function],
  state: { stop: true } }
--- Output: O
